nav2_behaviors/src/behavior_server.cpp

```diff
@@ -124,6 +123,72 @@ BehaviorServer::loadBehaviorPlugins()
   return true;
 }
 
+void BehaviorServer::configureBehaviorPlugins()
+{
+  auto node = shared_from_this();
+
+  for (size_t i = 0; i != behavior_ids_.size(); i++) {
+    behaviors_[i]->configure(
+      node,
+      behavior_ids_[i],
+      tf_,
+      local_collision_checker_,
+      global_collision_checker_);
+  }
+}
+
+void BehaviorServer::setupResourcesForBehaviorPlugins()
+{
+  std::string local_costmap_topic, global_costmap_topic;
+  std::string local_footprint_topic, global_footprint_topic;
+  std::string robot_base_frame;
+  double transform_tolerance;
+  get_parameter("local_costmap_topic", local_costmap_topic);
+  get_parameter("global_costmap_topic", global_costmap_topic);
+  get_parameter("local_footprint_topic", local_footprint_topic);
+  get_parameter("global_footprint_topic", global_footprint_topic);
+  get_parameter("transform_tolerance", transform_tolerance);
+  get_parameter("robot_base_frame", robot_base_frame);
+
+  bool need_local_costmap = false;
+  bool need_global_costmap = false;
+  for (const auto & behavior : behaviors_) {
+    if (behavior->getResourceInfo() == nav2_core::CostmapInfoType::BOTH) {
```

Comment:
`auto & type = behavior->getResourceInfo()` then `type` can be used in the if statements `if (type == BOTH)`, etc.